tests/test_penalty.py

```diff
@@ -1127,3 +1129,56 @@ def test_penalty_track_markers_with_nan(penalty_origin, value):
         np.array(res),
         expected,
     )
+
+
+@pytest.mark.parametrize("node", [Node.ALL, Node.INTERMEDIATES, Node.START, Node.MID, Node.PENULTIMATE, Node.END,
+                                  Node.TRANSITION])
+@pytest.mark.parametrize("ns", [10, 11])
+def test__get_node(node, ns):
+
+    NLP = nlp()
+    NLP.ns = ns
+    NLP.X = np.linspace(0, -10, ns+1)
+    NLP.U = np.linspace(10, 19, ns)
+
+    pn = []
+    penalty = PenaltyOption(pn)
+    penalty.node = node
+
+    if node == Node.MID and ns % 2 != 0:
+        with pytest.raises(ValueError, match="Number of shooting points must be even to use MID"):
+            t, x, u = PenaltyFunctionAbstract._get_node(NLP, penalty)
+    elif node == Node.TRANSITION:
+        with pytest.raises(RuntimeError, match=" is not a valid node"):
+            t, x, u = PenaltyFunctionAbstract._get_node(NLP, penalty)
+    else:
+        t, x, u = PenaltyFunctionAbstract._get_node(NLP, penalty)
+
+    x_expected = NLP.X
+    u_expected = NLP.U
+
+    if node == Node.ALL:
+        np.testing.assert_almost_equal(t, [i for i in range(ns + 1)])
+        np.testing.assert_almost_equal(np.array(x), np.linspace(0, -10, ns + 1))
+        np.testing.assert_almost_equal(np.array(u), np.linspace(10, 19, ns))
+    elif node == Node.INTERMEDIATES:
+        np.testing.assert_almost_equal(t, [i for i in range(1, ns-1)])
+        np.testing.assert_almost_equal(np.array(x), x_expected[1:ns-1])
+        np.testing.assert_almost_equal(np.array(u), u_expected[1:ns-1])
+    elif node == Node.START:
+        np.testing.assert_almost_equal(t, [0])
+        np.testing.assert_almost_equal(np.array(x), x_expected[0])
+        np.testing.assert_almost_equal(np.array(u), u_expected[0])
+    elif node == Node.MID and ns % 2 == 0:
+        np.testing.assert_almost_equal(t, [ns // 2])
+        np.testing.assert_almost_equal(np.array(x), x_expected[ns // 2])
+        np.testing.assert_almost_equal(np.array(u), u_expected[ns // 2])
+    elif node == Node.PENULTIMATE:
+        np.testing.assert_almost_equal(t, [ns - 1])
+        np.testing.assert_almost_equal(np.array(x), x_expected[-2])
+        np.testing.assert_almost_equal(np.array(u), u_expected[-1])
+    elif node == Node.END:
+        np.testing.assert_almost_equal(t, [ns])
+        np.testing.assert_almost_equal(np.array(x), x_expected[ns])
+        np.testing.assert_almost_equal(u, [])
+
```

Comment:
Please add a `else` that raises an error if it ever get call. Otherwise your test function will silently fail for all the nodes that are not comprise in your `if`.